dvc/remote/ssh/connection.py

```diff
@@ -135,24 +135,28 @@ def walk(self, directory, topdown=True):
 
         self._sftp_connect()
 
+        dirs = []
+        nondirs = []
         try:
             dir_entries = self._sftp.listdir_attr(directory)
+            for entry in dir_entries:
+                name = entry.filename
+                if S_ISDIR(entry.st_mode):
+                    dirs.append(name)
+                else:
+                    nondirs.append(name)
         except IOError as exc:
-            raise DvcException(
-                "couldn't get the '{}' remote directory files list".format(
-                    directory
-                ),
-                cause=exc,
-            )
-
-        dirs = []
-        nondirs = []
-        for entry in dir_entries:
-            name = entry.filename
-            if S_ISDIR(entry.st_mode):
-                dirs.append(name)
+            # In Windows some directories are protected by the OS. Skipped!
+            if os.name == "nt":
```

Comment:
This is a pretty strange check. So what exactly raises that error? listdir_attr? If so, why wrap the whole hoop in this try-except? Also, i don't think we should ignore EPERM like this, since it would be an unexpected behavior for the user. If he has supplied a remote with protected directories we should raise an error and let him know about it, and not hide it away.

Reply:
Yes. The above answered applied to this concern too. I dropped this in next patch